demos/encrypted-vm-image/README.md

```diff
@@ -0,0 +1,120 @@
+# Trusted startup of Encrypted VM image
+
+This demo shows how one can start up a [confidential virtual machine (CVM)](https://cloud.google.com/compute/confidential-vm/docs/about-cvm) whose VM image is protected with JinDisk. It provides tools that (1) can convert a plain virtual machine (VM) image into a JinDisk-protected one and (2) can use a JinDisk-protected VM image to start up a confidential VM. During the VM startup, a customizable process of remote attestation is triggered to fetch the root key that can unlock the JinDisk-protected VM image.
+
+The demo presents how to build and deploy a JinDisk-encrypted image on both Intel TDX and AMD SEV, while users can experiment with the demo without any VM-based TEE support.
+
+## High-level Workflow
+
+![](./workflow.jpeg)
+
+The workflow of this demo can be described as the following steps.
+
+- Step 1: Create a JinDisk-protected VM image from a base image. Specifically, a new VM image that includes an EFI partition, a boot partition, and the most crucial one - a root partition is created. The root partition (RootFS) should be transformed into the JinDisk format using a predetermined disk encryption key. Necessary components (initramfs hooks, RA scripts, and other software dependencies) will be included into the new image to enable the subsequent guest VM startup.
+
+- Step 2: The guest owner uses a TEE-supported Hypervisor (such as QEMU/KVM) to initiate a secure VM. Here, the *Guest Owner* is the client in a VM-based TEE environment that would like to use the confidential computing cloud.
+
+- Step 3, 4, and 5: The hooks inside the initramfs request the attestation report and sent it to the guest owner/key server when the kernel is booted. If the verification is passed, a trusted communication channel can be built and the key to decrypt the root partition is retrieved by the guest VM. The initramfs hooks then decrypt the guest VM's root filesystem with the key.
+
+## Step-by-step Instructions
+
+A step-by-step guide is given here to show how the guest owner can create the JinDisk-protected image and unlock the image during startup. All commands in this guide are expected to be executed on the host machine if not specified.
```

Comment:
I suggest rewriting the last sentence to the following form.

```
All commands in this guide are to be executed on the host machine if not specified otherwise.
```